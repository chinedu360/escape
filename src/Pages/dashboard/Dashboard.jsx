// Import necessary modules and components
import { useDispatch, useSelector } from "react-redux"; // Import useDispatch and useSelector from Redux for managing state
import { Modal, TableComp } from "../../components"; // Import custom components: Modal and TableComp
import { useEffect } from "react"; // Import useEffect hook for side effects
import {
  postData,
  userData,
  commentData,
} from "../../features/postSlice/postSlice"; // Import asynchronous thunks from postSlice

// Define the Dashboard component
const Dashboard = () => {
  const dispatch = useDispatch(); // Initialize the dispatch function from Redux for dispatching actions
  const { isLoading, comment, errorMessage, post, user } = useSelector(
    (store) => store.post
  ); // Use the useSelector hook to access state from the Redux store

  useEffect(() => {
    // Use the useEffect hook for handling side effects
    dispatch(postData()); // Dispatch the postData action to fetch post data
    dispatch(userData()); // Dispatch the userData action to fetch user data
    dispatch(commentData()); // Dispatch the commentData action to fetch comment data
  }, [dispatch]); // Run this effect when the dispatch function changes

  return (
    <div className="">
      <h2 className="ml-2 font-bold text-[25px] my-4">Welcome Back,</h2>
      {/* Render a welcome message */}
      <Modal
        length={post?.length}
        authorLen={user?.length}
        comment={comment?.length}
      />
      {/* Render the Modal component with information about post, user, and comments */}
      <div className="flex md:flex-row flex-col">
        <div className="w-[100%] md:w-[100%] space-y-4">
          <TableComp
            variant="striped"
            title="My Articles"
            titleName="Article Name"
            loading={isLoading}
            data={post}
            err={errorMessage}
            showTable="hidden"
          />
          {/* Render a table component displaying articles */}
        </div>
        <div className="w-[100%] md:w-[30%] space-y-4">
          <TableComp
            title="Best Authors"
            display={{ base: "none" }}
            titleName="Authors Name"
            show="hidden"
            data={user}
            err={errorMessage}
          />
          {/* Render a table component displaying authors */}
        </div>
      </div>
    </div>
  );
};

export default Dashboard; // Export the Dashboard component for use in the application
