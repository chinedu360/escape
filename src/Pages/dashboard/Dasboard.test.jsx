import { render, screen } from "@testing-library/react";
import Dashboard from "./Dashboard";
import { MemoryRouter } from "react-router-dom";
import { describe, it, expect } from "vitest";
import TestRenderer from "react-test-renderer";

// Import the necessary dependencies for creating a mocked Redux store
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import rootReducer from "../../features/postSlice/postSlice"; //

describe("testing the Dashboard", () => {
  it("renders Dashboard component", () => {
    // Create a mocked Redux store using configureStore
    const store = configureStore({
      reducer: rootReducer, // Replace with your root reducer
    });

    render(
      <MemoryRouter>
        <Provider store={store}>
          <Dashboard />
        </Provider>
      </MemoryRouter>
    );

    //Building assumption varaibles
    const titleElement = screen.getByText("Welcome Back,");

    //running expectation
    expect(titleElement).toBe(titleElement);
  });

  it("test the children inside the Dashboard component", () => {
    // Create a mocked Redux store using configureStore
    const store = configureStore({
      reducer: rootReducer, // Replace with your root reducer
    });

    const dashboard = TestRenderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <Dashboard />
        </Provider>
      </MemoryRouter>
    ).toJSON();

    // Dashboard containers
    expect(dashboard.type);
    expect(dashboard.children[0].type).toBe("h2");
    expect(dashboard.children[1].type).toBe("div");
    expect(dashboard.children[2].type).toBe("div");
  });

  it("Snapshot", () => {
    // Create a mocked Redux store using configureStore
    const store = configureStore({
      reducer: rootReducer, // Replace with your root reducer
    });

    const dashBoard = TestRenderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <Dashboard />
        </Provider>
      </MemoryRouter>
    ).toJSON();

    expect(dashBoard).toMatchSnapshot();
  });
});

// test("navigates to dashboard on button click", () => {
//   const mock = vi.fn();
//   mock.mockImplementation("react-router-dom", () => ({
//     ...vi.importActual("react-router-dom"),
//     useNavigate: () => mock,
//   }));

//   render(
//     <MemoryRouter>
//       <Landing />
//     </MemoryRouter>
//   );

//   const buttonElement = screen.getByText("Read More");
//   expect(buttonElement).toBeInTheDocument();

//   buttonElement.click();

//   // Make assertions on mock
//   // Make assertions on mock
//   // expect(mock.calls?.length).toBe(1); // Check the number of calls
//   expect(mock?.calls[0][0]).toBe("/dashboard"); // Check the first call arguments
// });
