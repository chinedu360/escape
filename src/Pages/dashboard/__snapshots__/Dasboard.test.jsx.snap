// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`testing the Dashboard > Snapshot 1`] = `
<div
  className=""
>
  <h2
    className="ml-2 font-bold text-[25px] my-4"
  >
    Welcome Back,
  </h2>
   
  <div
    className="m-2"
  >
    <div
      className="css-1ghed9e"
    >
      <div
        className="border-red-300 border-[2px] rounded-md css-170zrko"
      >
        <div
          className="css-1bwftuo"
        />
        <div
          className="css-1p2ux4r"
        >
          My Articles
        </div>
      </div>
      <div
        className="border-green-600 border-[2px] rounded-md css-1n1mkoi"
      >
        <div
          className="css-daheo5"
        />
        <div
          className="css-1p2ux4r"
        >
          Our Authors
        </div>
      </div>
      <div
        className="border-teal-500 border-[2px] rounded-md css-1n1mkoi"
      >
        <div
          className="css-h53m4y"
        />
        <div
          className="css-1p2ux4r"
        >
          Comments
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex md:flex-row flex-col"
  >
    <div
      className="w-[100%] md:w-[100%] space-y-4"
    >
      <div
        className="m-2 rounded-lg bg-white shadow-lg h-70vh overflow-y-scroll"
      >
        <h2
          className="m-2 font-semibold text-20px"
          style={
            {
              "backgroundColor": "white",
              "position": "sticky",
              "top": 0,
              "zIndex": 1,
            }
          }
        >
          My Articles
           
        </h2>
        <div
          className="chakra-table__container css-1gc4pzd"
        >
          <table
            className="chakra-table css-0"
          >
            <thead
              className="css-0"
            >
              <tr
                className="css-0"
              >
                <th
                  className="css-0"
                >
                   
                  Article Name
                </th>
                 
                <th
                  className="css-0"
                >
                  Description
                </th>
                 
              </tr>
            </thead>
            <tbody
              className="css-0"
            />
          </table>
        </div>
        <div
          className="undefined flex p-5 items-center cursor-pointer space-x-1"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M427 234.625H167.296l119.702-119.702L256 85 85 256l171 171 29.922-29.924-118.626-119.701H427v-42.75z"
            />
          </svg>
           
          <p>
            Back
          </p>
           
        </div>
      </div>
    </div>
    <div
      className="w-[100%] md:w-[30%] space-y-4"
    >
      <div
        className="m-2 rounded-lg bg-white shadow-lg h-70vh overflow-y-scroll"
      >
        <h2
          className="m-2 font-semibold text-20px"
          style={
            {
              "backgroundColor": "white",
              "position": "sticky",
              "top": 0,
              "zIndex": 1,
            }
          }
        >
          Best Authors
           
        </h2>
        <div
          className="chakra-table__container css-1gc4pzd"
        >
          <table
            className="chakra-table css-0"
          >
            <thead
              className="css-0"
            >
              <tr
                className="css-0"
              >
                <th
                  className="css-0"
                >
                   
                  Authors Name
                </th>
                 
                <th
                  className="css-v2jh1l"
                >
                  Description
                </th>
                 
              </tr>
            </thead>
            <tbody
              className="css-0"
            />
          </table>
        </div>
        <div
          className="hidden flex p-5 items-center cursor-pointer space-x-1"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M427 234.625H167.296l119.702-119.702L256 85 85 256l171 171 29.922-29.924-118.626-119.701H427v-42.75z"
            />
          </svg>
           
          <p>
            Back
          </p>
           
        </div>
      </div>
    </div>
  </div>
</div>
`;
